### DISCUSSION TITLE
Subscriptions using graphql-ws transport sub protocol not working

### DISCUSSION THREAD

**Discussion by Argeus** [graphql-java-kickstart/graphql-spring-boot#816] [Q&A]
Hey there! Been using this fantastic library for a while now but just recently tried to get subscriptions working and it strangely fails...

Been trying to debug why we can't get the new preferred transport for subscriptions in Apollo to work. I think I have found it.

The ApolloSubscriptionProtocolFactory sets it's type as "graphql-ws" but the protocol sent over the header from Apollo is "graphql-transport-ws". This is the proper name for the protocol as specified here:
https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md

This leads to subscriptions not working with the current library that Apollo recommends.

`public ApolloSubscriptionProtocolFactory(
      GraphQLObjectMapper objectMapper,
      GraphQLSubscriptionInvocationInputFactory invocationInputFactory,
      GraphQLInvoker graphQLInvoker,
      Collection<ApolloSubscriptionConnectionListener> connectionListeners,
      Duration keepAliveInterval) {
    super("graphql-ws");`

**Comment by ackmanx** [comment]
Guessing being https://github.com/apollographql/subscriptions-transport-ws was supported by Apollo that's where that protocol name comes from. 

I'm also trying to figure this out, how to configure the server to use `graphql-transport-ws` protocol instead
